Documentation/gettingstarted/cli-download.rst

```diff
@@ -16,10 +16,11 @@ various features (e.g. clustermesh, Hubble).
 
     .. code-block:: shell-session
 
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/latest/download/cilium-darwin-amd64.tar.gz{,.sha256sum}
-      shasum -a 256 -c cilium-darwin-amd64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-darwin-amd64.tar.gz /usr/local/bin
-      rm cilium-darwin-amd64.tar.gz{,.sha256sum}
+      export ARCH=$(uname -p)
```

Comment:
On my x86_64 Mac this will print `i386`, not `amd64` as I'd expect.

Depending on the terminal and shell the command is run (i.e. whether they are universal binaries), this might also return `x86_64` on an M1 Mac. See e.g. https://github.com/golang/go/issues/50643 where this caused problems.

Unfortunately it looks like there is no easy single-line shell command on macOS to cover all these cases 😞

Reply:
Could something [like this](https://apple.stackexchange.com/a/352770) help? I don't have a macOS so can't check if it works, and no idea what values you may obtain.

Reply:
on my m1 mac, uname -p returns `arm`, I was expecting to have arm64 🤔

Reply:
This is the output I get on the x86_64 Mac:

<details>
<summary>command output</summary>

```
 % sysctl -a machdep.cpu
machdep.cpu.address_bits.physical: 39
machdep.cpu.address_bits.virtual: 48
machdep.cpu.arch_perf.events_number: 7
machdep.cpu.arch_perf.fixed_number: 3
machdep.cpu.arch_perf.fixed_width: 48
machdep.cpu.arch_perf.number: 4
machdep.cpu.arch_perf.version: 4
machdep.cpu.arch_perf.width: 48
machdep.cpu.arch_perf.events: 0
machdep.cpu.cache.L2_associativity: 4
machdep.cpu.cache.linesize: 64
machdep.cpu.cache.size: 256
machdep.cpu.mwait.extensions: 3
machdep.cpu.mwait.linesize_max: 64
machdep.cpu.mwait.linesize_min: 64
machdep.cpu.mwait.sub_Cstates: 286531872
machdep.cpu.thermal.ACNT_MCNT: 1
machdep.cpu.thermal.core_power_limits: 1
machdep.cpu.thermal.dynamic_acceleration: 1
machdep.cpu.thermal.energy_policy: 1
machdep.cpu.thermal.fine_grain_clock_mod: 1
machdep.cpu.thermal.hardware_feedback: 0
machdep.cpu.thermal.invariant_APIC_timer: 1
machdep.cpu.thermal.package_thermal_intr: 1
machdep.cpu.thermal.sensor: 1
machdep.cpu.thermal.thresholds: 2
machdep.cpu.tlb.data.small: 64
machdep.cpu.tlb.data.small_level1: 64
machdep.cpu.tlb.inst.large: 8
machdep.cpu.tsc_ccc.denominator: 2
machdep.cpu.tsc_ccc.numerator: 134
machdep.cpu.xsave.extended_state: 31 832 1088 0
machdep.cpu.xsave.extended_state1: 15 832 256 0
machdep.cpu.brand: 0
machdep.cpu.brand_string: Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz
machdep.cpu.core_count: 2
machdep.cpu.cores_per_package: 8
machdep.cpu.extfamily: 0
machdep.cpu.extfeature_bits: 1241984796928
machdep.cpu.extfeatures: SYSCALL XD 1GBPAGE EM64T LAHF LZCNT PREFETCHW RDTSCP TSCI
machdep.cpu.extmodel: 8
machdep.cpu.family: 6
machdep.cpu.feature_bits: 9221959987971750911
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
machdep.cpu.leaf7_feature_bits: 43804591 0
machdep.cpu.leaf7_feature_bits_edx: 3154128384
machdep.cpu.leaf7_features: RDWRFSGS TSC_THREAD_OFFSET SGX BMI1 AVX2 SMEP BMI2 ERMS INVPCID FPU_CSDS MPX RDSEED ADX SMAP CLFSOPT IPT MDCLEAR TSXFA IBRS STIBP L1DF ACAPMSR SSBD
machdep.cpu.logical_per_package: 16
machdep.cpu.max_basic: 22
machdep.cpu.max_ext: 2147483656
machdep.cpu.microcode_version: 236
machdep.cpu.model: 142
machdep.cpu.processor_flag: 4
machdep.cpu.signature: 526057
machdep.cpu.stepping: 9
machdep.cpu.thread_count: 4
machdep.cpu.vendor: GenuineIntel
```

</details>

Unfortunately, I don't have a M1 Mac to compare.

From a quick glance we might be able to somehow deduce the archiecture from `machdep.cpu.brand`, `machdep.cpu.family` or `machdep.cpu.vendor`?

`sysctl hw.optional.x86_64` also looks interesting. I wonder whether there is something like `hw.optional.aarch64` on M1 Macs?

Reply:
here is the output for m1 mac :) 

```
$ sysctl -a machdep.cpu
machdep.cpu.brand_string: Apple M1 Max
machdep.cpu.core_count: 10
machdep.cpu.cores_per_package: 10
machdep.cpu.logical_per_package: 10
machdep.cpu.thread_count: 10
```

Reply:
> `sysctl hw.optional.x86_64` also looks interesting. I wonder whether there is something like `hw.optional.aarch64` on M1 Macs?

@sayboras checked this on an M1 Mac and there seems to be a `hw.optional.arm64` value there.

So we’d need some shell magic to set `ARCH=arm64` if `sysctl hw.optional.arm64` returns `1` and `ARCH=amd64` if `sysctl hw.optional.x86_64` returns `1`, respectively.